source-mongodb/pull.go

```diff
@@ -134,11 +190,39 @@ type capture struct {
 type captureState struct {
 	Resources map[string]resourceState `json:"resources,omitempty"`
 
+	ChangeStreamEmptyBeforeBackfill *bool `json:"change_stream_empty_before_backfill,omitempty"`
```

Comment:
I would rename this to `OplogEmptyBeforeBackfill` just because change stream is an API on top of oplog, and we check the oplog for being empty, not the change stream. To be very technical, we may have an empty change stream when tailing if there are no new events, but there might still be records in the oplog.